services/lab-service/lab-service/main.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018 Telstra Open Source
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import os
+import sys
+import time
+import signal
+import logging
+
+import requests
+from topology import Topology
+from lockkeeper import run_server
+
+logger = logging.getLogger()
+
+LAB_ID = os.environ.get("LAB_ID", 1)
+API_HOST = os.environ.get("API_HOST", 'lab-api.pendev:8288')
+
+
+if __name__ == '__main__':
+    url = "http://{}/api/{}/definition".format(API_HOST, LAB_ID)
+    topo_def = requests.get(url).json()
```

Comment:
Getting it via stdin should/can be much simpler.

Reply:
Nope. In case of stdin it should  take care about json escaping when pass through docker arguments\env, and json string may be quite big

Reply:
Why it can't be started via sh wrapper?

Reply:
This code invoked in container by request from containerized Api, that itself gets JSON definition through http from test framework. So, chain is: Test framework -> Api (docker) -> Docker daemon -> **Topology** (docker). 
From my point of view it's easier to ask Api for topology definition by http than passing it through docker daemon some other way.